Main.elm

```diff
@@ -73,30 +75,69 @@ setupControlsView address player =
 shipFieldView : Signal.Address Action -> Ship.Ship -> Html.Html
 shipFieldView address ship =
   if not ship.added then
-    Html.div []
+    Html.div [ Html.Attributes.style [("margin-bottom", "5px")] ]
     [ Html.input -- Ship Row Input
       [ Html.Attributes.value (toString (Ship.getRow ship))
       , Html.Events.on "input" Html.Events.targetValue (Signal.message address << SetupRowField ship.id)
       ]
       []
     , Html.input -- Ship Column Input
-      [ Html.Attributes.value (toString (Ship.getColumn ship))
+      [ Html.Attributes.style [("margin-right", "5px")]
+      , Html.Attributes.value (toString (Ship.getColumn ship))
       , Html.Events.on "input" Html.Events.targetValue (Signal.message address << SetupColumnField ship.id)
       ]
       []
-    , Html.input -- Ship Orientation
-      [ Html.Attributes.type' "radio"
-      , Html.Attributes.checked (ship.orientation == Ship.Horizontal)
-      , Html.Events.onClick address (SetupOrientationField ship.id)
-      ]
-      []
+      -- Ship orientation:
+    , orientationSwitch address ship.id ship.orientation
     , Html.button -- Add Ship
       [ Html.Events.onClick address (SetupAddShip ship.id)
       ]
-      []
+      [ Html.text "Add ship" ]
     ]
   else Html.div [] []
 
+orientationSwitch : Signal.Address Action -> Int -> Ship.Orientation -> Html.Html
+orientationSwitch address shipId orientation =
```

Comment:
The ui is more of a hook into the functionality. The functionality for changing a ships orientation (and many other functions) is independent from how the ui is implemented. For example, I see the ability to "CLICK" to add ships into the grid as a nice ui feature. That ui feature is going to use the same code for adding ships that the current ui uses.